keps/draft-20180426-security-conformance.md

```diff
@@ -0,0 +1,465 @@
+
+The criteria described below is based on a namespace with default settings.
+Like a newly created namespace, with only service account `default`.
+Namespaces with special/privileged settings, like `kube-system`,
+or special namespaces required by some system components
+(e.g. nginx ingress controller) are not covered.
+
+### Behavioral Criteria
+
+#### Config File Permissions
+
+  *Conformance Profile: Essential*
+
+  The configuration files for API server, controller manager, scheduler and kubelet should have the correct setup of ownership and file permissions.
+
+  - etcd data directory is owned and accessible only by the user running etcd;
+  - kubeconfig files for control plane is owned and accessible only by the user runs the control plane;
+  - private key files is owned and accessible only by the user who runs the process which consumes the key;
+  - kubeconfig for administration purpose is owned and accessible only by the admin user/group.
+
+
+  Note: the above statements exclude _root_ user who always have full access to the system.
+
+  See [Implementation Details](#implementation-details) below for test recommendation.
+
+#### API privileges of default service accounts
+
+  *Conformance Profile: Essential*
+
+  Every namespace has a service account named `default`.
+  Without explicit permission grant from RBAC RoleBinding or ClusterRoleBinding,
+  This service account has the minimum privilege to access Kubernetes API,
+  it's forbidden for the service account to `get/list/watch/create/update/patch/delete` any of API objects
+
+###### Test Recommendation
+
+  The test code consumes Kubernetes API as a serviceaccount/user with privileges:
+
+  - Create/Delete a namespace
+  - Impersonate the `default` service account in the new namespace
+
+
+  Test setup
+
+  - Create a new namespace
+  - Create a Pod without specifying `SecurityContext`, `ServiceAccountName`, without any volumes
+
+
+  Test steps  
+
+  - Impersonate the `default` service account to
+    `get/list/watch/create` the objects defined by `core` api group and get _forbidden_
+  - Impersonate the `default` service account to
+    `get/list/watch/create` the rbac objects: `Role/RoleBinding/ClusterRole/ClusterRoleBinding` and get _forbidden_
+  - Impersonate the `default` service account to
+    `update/patch` the created Pod by changing some fields, like a label, should get _forbidden_
+  - Impersonate the `default` service account to `delete` the Pod, and should get _forbidden_
+
+
+  Test teardown
+
+  - Delete the namespace
+
+#### Kubelet r/w API Access
+
+  *Conformance Profile: Essential*
+
+  _TODO: need more context_
+
+  Anonymous access to Kubelet r/w APIs should be denied as unauthorized.
+  Default service accounts should also be denied as forbidden accessing Kubelet r/w APIs.
+
+#### Privileged Containers
+
+  *Conformance Profile: Restricted*
+
+  Pod creation with `privileged` set to `true` in `SecurityContext` (Pod or container level)
```

Comment:
It's true today, but not because of `privileged` in `SecurityContext`. Once PodSecurityPolicy admission control is enabled, no Pod will be created if there's no RBAC rolebinding/clusterrolebinding is associated with the Pod service account.